gems/pending/util/miq_winrm.rb

```diff
@@ -17,17 +18,33 @@ def connect(options = {})
     validate_options(options)
     @uri        = build_uri
     @connection = raw_connect(@username, @password, @uri)
+    @operations = 0
     @executor   = @connection.create_executor
   end
 
+  def close
+    @executor.close
+    @executor = nil
+  end
+
   def run_powershell_script(script)
     $log.debug "Running powershell script on #{hostname} as #{username}:\n#{script}" unless $log.nil?
-    @executor.run_powershell_script(script)
+    executor.run_powershell_script(script)
   rescue WinRM::WinRMAuthorizationError
     $log.info "Error Logging In to #{hostname} using user \"#{username}\"" unless $log.nil?
     raise
   end
 
+  def executor
+    @operations += 1
+    return @executor if @operations < MAX_WINRM_CMDS
+    $log.debug "MiqWinRM: #{@operations} operations on executor - restarting."
```

Comment:
I'm not sure vmdb/logging is available from the gems/pending code yet.  On the TODO list :/